app/controllers/wallets_controller.rb

```diff
@@ -0,0 +1,37 @@
+class WalletsController < ApplicationController
+  before_action :logged_in_user
+  def new
+    @wallet = Wallet.new
+  end
+
+  def create
+    if create_wallet
+      flash[:success] = t ".success"
+      redirect_to wallets_url
+    else
+      flash.now[:danger] = t ".failure"
+      render :new
+    end
+  end
+
+  def show
+    @wallet = Wallet.find_by id: params[:id]
+    return @wallet if current_user? @wallet.user_id
+
+    flash[:danger] = t ".unauthorization"
+    redirect_to wallets_url
+  end
+
+  def index
+    @wallets = Wallet.newest.where user_id: current_user.id
```

Comment:
dùng ưu thế của association đi em ei
```suggestion
    @wallets = current_user.wallets.newest
```